src/include/catalog/sequence_catalog.h

```diff
@@ -0,0 +1,156 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// sequence_catalog.h
+//
+// Identification: src/include/catalog/sequence_catalog.h
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// pg_trigger
+//
+// Schema: (column offset: column_name)
+// 0: oid (pkey)
+// 1: sqdboid   : database_oid
+// 2: sqname    : sequence_name
+// 3: sqinc     : seq_increment
+// 4: sqmax     : seq_max
+// 5: sqmin     : seq_min
+// 6: sqstart   : seq_start
+// 7: sqcycle   : seq_cycle
+// 7: sqval     : seq_value
+//
+// Indexes: (index offset: indexed columns)
+// 0: oid (primary key)
+// 1: (sqdboid, sqname) (secondary key 0)
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+#include <utility>
+#include <mutex>
+
+#include "catalog/abstract_catalog.h"
+#include "catalog/catalog_defaults.h"
+
+#define SEQUENCE_CATALOG_NAME "pg_sequence"
+
+namespace peloton {
+
+namespace concurrency {
+class TransactionContext;
+}
+
+namespace catalog {
+
+class SequenceCatalogObject {
+ public:
+  SequenceCatalogObject(oid_t seqoid, const std::string &name,
+                        const int64_t seqstart, const int64_t seqincrement,
+                        const int64_t seqmax, const int64_t seqmin,
+                        const bool seqcycle, const int64_t seqval,
+                        concurrency::TransactionContext *txn)
+      : seq_oid(seqoid),
+        seq_name(name),
+        seq_start(seqstart),
+        seq_increment(seqincrement),
+        seq_max(seqmax),
+        seq_min(seqmin),
+        seq_cycle(seqcycle),
+        txn_(txn),
+        seq_curr_val(seqval){};
+
+  oid_t seq_oid;
+  std::string seq_name;
+  int64_t seq_start;      // Start value of the sequence
+  int64_t seq_increment;  // Increment value of the sequence
+  int64_t seq_max;        // Maximum value of the sequence
+  int64_t seq_min;        // Minimum value of the sequence
+  int64_t seq_cache;      // Cache size of the sequence
+  bool seq_cycle;         // Whether the sequence cycles
+  concurrency::TransactionContext *txn_;
+
+  std::mutex sequence_mutex;  // mutex for all operations
```

Comment:
Maybe unnecessary for the current implementation.

Reply:
Is this mutex being used anywhere? Also, you should use `common::synchronization::DirtyMutexLatch`

https://github.com/cmu-db/peloton/blob/master/src/include/common/synchronization/mutex_latch.h